dawn/src/dawn/Optimizer/TemporaryHandling.cpp

```diff
@@ -70,19 +71,23 @@ void promoteLocalVariableToTemporaryField(iir::StencilInstantiation* instantiati
     DAWN_ASSERT_MSG(!varDeclStmt->isArray(), "cannot promote local array to temporary field");
 
     auto fieldAccessExpr = std::make_shared<iir::FieldAccessExpr>(fieldname);
-    instantiation->getMetaData().insertExprToAccessID(fieldAccessExpr, accessID);
+    fieldAccessExpr->getData<iir::IIRAccessExprData>().AccessID = std::make_optional(accessID);
     auto assignmentExpr =
         std::make_shared<iir::AssignmentExpr>(fieldAccessExpr, varDeclStmt->getInitList().front());
     auto exprStmt = iir::makeExprStmt(assignmentExpr);
 
     // Replace the statement
     exprStmt->getData<iir::IIRStmtData>().StackTrace =
         oldStatement->getData<iir::IIRStmtData>().StackTrace;
+    exprStmt->getData<iir::IIRStmtData>().CallerAccesses =
+        oldStatement->getData<iir::IIRStmtData>().CallerAccesses;
```

Comment:
factor out `exprData` somewhere above and access through that?

Reply:
Since `oldStatement` is going to be replaced, better to `std::move` the data:
```
exprStmt->getData<iir::IIRStmtData>() = std::move(oldStatement->getData<iir::IIRStmtData>());
```